serde_arrow/src/internal/deserialization/date64_deserializer.rs

```diff
@@ -43,11 +43,39 @@ impl<'a> Date64Deserializer<'a> {
         };
 
         if self.is_utc {
+            Ok(self.format_utc(date_time))
```

Comment:
Add `format_with_suffix` and call as `format_with_suffix(.., "Z")` and `format_with_suffix(.., "")`